Assets/Mirror/Runtime/NetworkClient.cs

```diff
@@ -222,6 +222,7 @@ public void OnAuthenticated(NetworkConnectionToServer conn)
         /// </summary>
         public void Disconnect()
         {
+            connectState = ConnectState.Disconnected;
```

Comment:
I get it,  but this is a little more complicated than that.

Right now when you StartClient,  the Connection remains null until it is accepted,  which could take potentially a long time.  If someone calls disconnect,  then this will mark the connection as disconnected,  but the connection might still succeed after that.

Reply:
wouldn't that be covered by the next line? Connection?.Disconnect();

I did a test where the client connects before the server comes up. The connection never completes. So the only chance for it to succeed after a period of time would be if there is ALOT of latency involved.

I also did a test with the new line removed. If a connection is established the normal flow and NetManHUD work fine.